File: mobile/lib/mappers/user_mapper.dart
import 'package:quizora/dto/user_dto.dart';
import 'package:quizora/models/identity.dart';
import 'package:quizora/models/user.dart';

class UserMapper {
  static User fromDto(UserDto dto) {
    return User(
      email: dto.email,
      firstName: dto.firstName,
      lastName: dto.lastName,
      role: dto.role,
      identity: Identity.fromJson(dto.identity.toJson()),
    );
  }
}

File: frontend/src/mappers/user.mapper.ts
import type { UserDto } from "@/types/user.type";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const buildUserDto = (auth0User: any): UserDto => {
  const [provider, providerUserId] = auth0User.sub.split("|");
  console.log("Auth0 User:", auth0User);
  const identity = {
    provider,
    providerUserId,
    authId: auth0User.sub,
  };

  return {
    email: auth0User.email,
    firstName: auth0User.given_name || "",
    lastName: auth0User.family_name || "",
    picture: auth0User.picture,
    role: "student",
    identity,
    provider,
  };
};
